src/players/Player.js

```diff
@@ -0,0 +1,47 @@
+import { MESSAGE } from '../constants/constants.js';
+import View from '../view/View.js';
+
+export default class Player {
+
+  async inputNumber() {
+    const userInput = await View.readInput(MESSAGE.input);
+
+    if (Player.isValidNumber(userInput)) {
+      this._playerNumber = userInput.split('').map(num => parseInt(num, 10));
+    }
+  }
+
+  getJudgeResult(computer) {
+    const result = computer.judgeResult(this._playerNumber);
+    return result;
+  }
+
+  static isValidNumber(userInput) {
+    const userInputToArray = userInput.split('');
+
+    if (userInputToArray.length !== 3) {
+      throw new Error(MESSAGE.error);
+    }
+
+    if (userInputToArray.some(char => isNaN(parseInt(char, 10)))) {
+      throw new Error(MESSAGE.error);
+    }
+
+    const numericInput = userInputToArray.map(num => parseInt(num, 10));
+
+    if (numericInput.includes(0)) {
+      throw new Error(MESSAGE.error);
+    }
+
+    if (Player.hasDuplicates(numericInput)) {
+      throw new Error(MESSAGE.error);
+    }
+
+    return true;
+  }
+
+  static hasDuplicates(userNumber) {
+    const uniqueNumbers = new Set(userNumber);
+    return uniqueNumbers.size !== userNumber.length;
+  }
+}
```

Comment:
매핑을 `Array.from()` 으로 개선한 후, 바로 `Set` 객체와 Set 객체의 `has()` 함수를 사용해서
`hasDuplicates(userNumber)` 함수를 제거해보는 쪽으로 코드를 개선해봤습니다

```suggestion
    const numericInput = new Set(Array.from(userInputToArray, Number));
    
    if (numericInput.has(0)) {
      throw new Error(MESSAGE.error);
    }

    return true;
  }
}
```

Reply:
상당히 좋은 방법이네요. 다른 곳에선 중복검사를 set으로 했는데,
왜 여기서는 생각하지 못했을까요 .
감사합니다~